services/src/main/java/io/scalecube/services/LocalServiceInstance.java

```diff
@@ -55,11 +55,7 @@ public Object invoke(Message message) throws Exception {
       if (method.getParameters().length == 0) {
         result = method.invoke(serviceObject);
       } else if (method.getParameters()[0].getType().isAssignableFrom(Message.class)) {
-        if (message.data().getClass().isAssignableFrom(Message.class)) {
-          result = method.invoke(serviceObject, (Message) message.data());
-        } else {
-          result = method.invoke(serviceObject, message);
-        }
+        result = method.invoke(serviceObject, (Message) message.data());
```

Comment:
Not sure that I know the logic supposed to be here, but is it really that message.data() is always an Message here?